cms/djangoapps/contentstore/views/tests/test_transcript_settings.py

```diff
@@ -0,0 +1,167 @@
+import ddt
+import json
+from mock import Mock, patch
+
+from django.test.testcases import TestCase
+
+from contentstore.tests.utils import CourseTestCase
+from contentstore.utils import reverse_course_url
+from contentstore.views.transcript_settings import validate_transcript_credentials
+
+
+@ddt.ddt
+@patch(
+    'openedx.core.djangoapps.video_config.models.VideoTranscriptEnabledFlag.feature_enabled',
+    Mock(return_value=True)
+)
+class TranscriptCredentialsTest(CourseTestCase):
+    """
+    Tests for transcript credentials handler.
+    """
+    VIEW_NAME = 'transcript_credentials_handler'
+
+    def get_url_for_course_key(self, course_id):
+        return reverse_course_url(self.VIEW_NAME, course_id)
+
+    def test_302_with_anonymous_user(self):
+        """
+        Verify that redirection happens in case of unauthorized request.
+        """
+        self.client.logout()
+        transcript_settings_url = self.get_url_for_course_key(self.course.id)
+        response = self.client.post(transcript_settings_url, content_type='application/json')
+        self.assertEqual(response.status_code, 302)
+
+    def test_405_with_not_allowed_request_method(self):
+        """
+        Verify that 405 is returned in case of not-allowed request methods.
+        Allowed request methods include POST.
+        """
+        transcript_settings_url = self.get_url_for_course_key(self.course.id)
+        response = self.client.get(transcript_settings_url, content_type='application/json')
+        self.assertEqual(response.status_code, 405)
+
+    def test_404_with_feature_disabled(self):
+        """
+        Verify that 404 is returned if the corresponding feature is disabled.
+        """
+        transcript_settings_url = self.get_url_for_course_key(self.course.id)
+        with patch('openedx.core.djangoapps.video_config.models.VideoTranscriptEnabledFlag.feature_enabled') as feature:
+            feature.return_value = False
+            response = self.client.post(transcript_settings_url, content_type='application/json')
+            self.assertEqual(response.status_code, 404)
+
+    @ddt.data(
+        (
+            {
+                'provider': 'invalid_provider',
+                'api_key': '1234'
+            },
+            True,
+            400
+        ),
+        (
+            {
+                'provider': '3PlayMedia',
+                'api_key': '11111',
+                'api_secret_key': '44444'
+            },
+            False,
+            400
+        ),
+        (
+            {
+                'provider': 'Cielo24',
+                'api_key': '12345',
+                'username': 'test_user'
+            },
+            True,
+            200
+        )
+    )
+    @ddt.unpack
```

Comment:
@Qubad786 We should also look for the scenarios for these case 

```
In case of:
    3PlayMedia - 'api_key' and 'api_secret_key' are required.
    Cielo24 - 'api_key' and 'username' are required.
```

Reply:
oh you have added tests below, NVM just saw that